### PR TITLE
feat: Add GitLab CI provenance (#6375)

### PR DESCRIPTION
**Pull request by wraithgar**
This is a first pass at provenance generation for GitLab CI.

This is based loosely off of existing GitLab provenance documents:
https://about.gitlab.com/blog/2022/11/30/achieve-slsa-level-2-compliance-with-gitlab/
https://gist.github.com/wlynch/c7fd8f53adc77d3c0ec82356e4d43cb5

### CODE REVIEW

**Comment by wraithgar** [comment]
Per @feelepxyz:

> I would like to hold off merging this until we've fixed up some of the links in the UI. Should get this done this week.

**Comment by wraithgar** [comment]
@wlynch This branch has been rebased against latest and had merge conflicts resolved.  We have a bit of time before we land the UI links for you to do a final check of this branch again after the rebase.

File: workspaces/libnpmpublish/lib/provenance.js [hunk 1]
```diff
@@ -1,69 +1,204 @@
...
 
-const BUILDER_ID = 'https://github.com/actions/runner'
-const BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
-const BUILD_TYPE_VERSION = 'v2'
+const GITHUB_BUILDER_ID = 'https://github.com/actions/runner'
+const GITHUB_BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gha'
+const GITHUB_BUILD_TYPE_VERSION = 'v2'
+
+const GITLAB_BUILD_TYPE_PREFIX = 'https://github.com/npm/cli/gitlab'
+const GITLAB_BUILD_TYPE_VERSION = 'v0alpha1'
```
Comment (wraithgar) [inline workspaces/libnpmpublish/lib/provenance.js:15]: There was a question as to if we wanted this to change to beta?

Comment (feelepxyz) [reply workspaces/libnpmpublish/lib/provenance.js:15]: Also seems ok if you want to keep as is @wlynch?

**Review by feelepxyz — APPROVED** [review]
The provenance UI for gitlab now renders working URLs: https://www.npmjs.com/package/@ps-testing/gitlab-npm-provenance#provenance

I'm 👍  on merging this into `latest` now 🎉

File: workspaces/libnpmpublish/lib/provenance.js (hunk 1 above)
Comment (wlynch) [reply workspaces/libnpmpublish/lib/provenance.js:15]: I'm fine with keeping this for now. I think there may be a few more tweaks coming to the provenance on the Fulcio side (https://github.com/sigstore/fulcio/pull/1206), but it shouldn't change anything w.r.t. npm. We can always rev this later.

**Pull request merged**